src/java/org/apache/cassandra/service/accord/AccordSafeCommandStore.java

```diff
@@ -246,28 +247,27 @@ private <O> O mapReduceForRange(Routables<?> keysOrRanges, Ranges slice, BiFunct
     {
         if (commandsForRanges == null)
             return accumulate;
+        CommandsForRanges cfr = commandsForRanges.current().slice(slice);
```

Comment:
this was found by `TopologyMixupTest`.  The `CommandsForRanges` shows every txn that could be found on *this* node, but we are processing a `Shard`, which is a smaller range.  So need to post process and `slice` to the shard's subset

Reply:
we do already.  That section you showed is actually

```
Routables<?> sliced = keysOrRanges.slice(slice, Routables.Slice.Minimal);
for (Key key : commandsForKeys.keySet()) {
  if (!sliced.contains(key)) continue;
```

we *could* slice the keys but this is a set, so we would first need to sort them, wrap them, then sort them... the contains check is slightly less work?